tools/test/LcovMerger/java/com/google/devtools/lcovmerger/Coverage.java

```diff
@@ -45,6 +46,25 @@ static Coverage merge(Coverage c1, Coverage c2) {
     return merged;
   }
 
+  static Coverage filterOutMatchingSources(
+      Coverage coverage, String[] regex) throws IllegalArgumentException {
+    if (coverage == null || regex == null) {
+      throw new IllegalArgumentException("Coverage and regex should not be null.");
+    }
+    if (regex.length == 0) {
+      return coverage;
+    }
+    Coverage filteredCoverage = new Coverage();
+    coverage.getAllSourceFiles().stream()
+            .filter(s -> !matchesAnyRegex(s.sourceFileName(), regex))
+            .forEach(s -> filteredCoverage.add(s));
```

Comment:
Is it OK that `forEach` does not preserve the iteration order?

Reply:
See my comment below (line 65). I think the same holds here: a C-style loop would be more efficient at the cost of slightly less readable code.

Reply:
I replaced it with iterator loop.